File: algorithms/171.Excel_Sheet_Column_Number/Excel_Sheet_Column_Number.java
<<<<<<< HEAD
public class Solution {
    public int titleToNumber(String s) {
        
    	// from 26 to 10
    	int result = 0;
    	
    	for (int i=0; i<s.length(); i++) {
    		result = ((int)s.charAt(i)-(int)'A'+1)+26*result;
    	}
    	
    	return result;
    }
=======
public class Solution {
    public int titleToNumber(String s) {
        
    	// from 26 to 10
    	int result = 0;
    	
    	for (int i=0; i<s.length(); i++) {
    		result = ((int)s.charAt(i)-(int)'A'+1)+26*result;
    	}
    	
    	return result;
    }
>>>>>>> 6200c8704614e918c8bfa5357c648dd1b4f7eb74
}

File: algorithms/171.Excel_Sheet_Column_Number/Excel_Sheet_Column_Number.py
class Solution:
    # @param s, a string
    # @return an integer
    def titleToNumber(self, s):
        return reduce(lambda x, y: 26 * x + ord(y) - ord('A') + 1, s, 0)

# debug
s = Solution()
print s.titleToNumber('AA')



class Solution:
    # @param s, a string
    # @return an integer
    def titleToNumber(self, s):
        result = 0
        for i in xrange(len(s)):
            result *= 26
            result += ord(s[i]) - ord('A') + 1
        return result

if __name__ == "__main__":
    print Solution().titleToNumber("AAAB")
